lang/cs/Org.Apache.REEF.Client/YARN/SecurityTokenWriter.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+using Org.Apache.REEF.Client.YARN.Parameters;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Client.YARN
+{
+    /// <summary>
+    /// Helper class for Security token.
+    /// </summary>
+    internal class SecurityTokenWriter
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(SecurityTokenWriter));
+
+        private const string SecurityTokenIdFile = "SecurityTokenId";
+        private const string SecurityTokenPwdFile = "SecurityTokenPwd";
+
+        private readonly IList<byte[]> _keys = new List<byte[]>();
+        private readonly IList<string> _pwds = new List<string>();
+
+        internal const string DefaultTokenKind = "NULL";
+        internal const string DefaultService = "NULL";
+
+        /// <summary>
+        /// Injectable constructor that accepts a set of serialized tokens.
+        /// Each serialized token string in the set is a serialized SecurityTokenInfo with JsonConvert 
+        /// </summary>
+        /// <param name="serializedTokenStrings"></param>
+        [Inject]
+        private SecurityTokenWriter([Parameter(typeof(SecurityTokenStrings))] ISet<string> serializedTokenStrings)
+        {
+            ParseTokens(serializedTokenStrings.ToList());
```

Comment:
Use `IEnumerable` in `ParseTokens` so you don't have to convert `Set` into `List` here